addon/services/activity-watcher.ts

```diff
@@ -0,0 +1,152 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+
+import { Observable } from '@upfluence/hyperevents/helpers/observable';
+import EventsService, { ResourceEvent, prefixPath } from '@upfluence/hyperevents/services/events-service';
+import ToastService, { ToastOptions } from '@upfluence/oss-components/services/toast';
+
+type NotificationEvent = {
+  resource: string;
+  payload: {
+    notification_type: string;
+    timestamp: number;
+    data: any;
+  };
+};
+
+type RenderedNotification = {
+  title: string;
+  message: string;
+  type: string;
+};
+
+function renderNotificationMessage(message: string): RenderedNotification {
+  return {
+    title: `<i class="toast-title__icon upf-icon upf-icon--messages"></i>`,
+    message: message,
+    type: 'info'
+  };
+}
+
+function renderNotificationErrorMessage(message: string): RenderedNotification {
+  return {
+    title: `<i class="fa fa-info-circle" aria-hidden="true"></i>`,
+    message: message,
+    type: 'error'
+  };
+}
+
+function renderNotificationMessageWithAvatar(avatarUrl: string, message: string): RenderedNotification {
+  return {
+    title: `<img class="toast-title__avatar" src="${avatarUrl}" />
+     <i class="toast-title__icon upf-icon upf-icon--messages"></i>`,
+    message: message,
+    type: 'info'
+  };
+}
+
+const toastOpts: ToastOptions = { timeout: 0 };
+
+export default class ActivityWatcher extends Service {
+  @service declare eventsService: EventsService;
+  @service declare toast: ToastService;
+
+  private declare _observer: Observable<ResourceEvent> | null;
+
+  watch(): void {
+    if (this._observer) {
+      return;
+    }
+
+    this._observer = this.eventsService.watch(prefixPath('/notification'));
+    this._observer.subscribe((evt: NotificationEvent) => {
+      this._processEvent(evt);
+    });
+  }
+
+  unwatch(): void {
+    if (!this._observer) {
+      return;
+    }
+
+    this._observer.unsubscribe();
+    this._observer = null;
+  }
+
+  private _processEvent(evt: NotificationEvent): void {
+    const notif = this._buildNotification(evt);
+
+    if (!notif) {
+      return;
+    }
+
+    switch (notif.type) {
+      case 'info':
+        this.toast.info(notif.message, notif.title, toastOpts);
+        break;
+      case 'error':
+        this.toast.error(notif.message, notif.title, toastOpts);
+        break;
+    }
+  }
+
+  private _buildNotification(evt: NotificationEvent): RenderedNotification | null {
```

Comment:
:/

How about defining a "dictionnary" that assigns the proper parameters to a specific method ?
Not easy suggesting something without trying it out, but I'm available if you want to have a look together :)

Reply:
Always happy to pair. Might give this a try :)